### PR TITLE
Migration for node hash invalidation

### PR DESCRIPTION
**Pull request by szoupanos**
This is a fix for issue #2131

### CODE REVIEW

**Comment by giovannipizzi** [comment]
There are a number of failing tests

File: aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    """Invalidating node hash - User should rehash nodes for caching"""
+
+    dependencies = [
+        ('db', '0014_add_node_uuid_unique_constraint'),
+    ]
+
+    operations = [
+        migrations.RunPython(notify_user, reverse_code=notify_user),
+        migrations.RunSQL(
+            """ DELETE FROM db_dbextra WHERE key='""" + _HASH_EXTRA_KEY + """';""",
```
Comment (sphuber) [inline aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py]: I think it is better to hard code the hash key. If at some point we change the hash key this migration will not work. Even though at that point we will have to add another migration, which would render this one obsolete. Still there would be a scenario possible where the change in key is done before the migration and then this migration is executed, thereby missing all the current hashes.

File: aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+from aiida.common.hashing import _HASH_EXTRA_KEY
+from aiida.cmdline.utils.echo import echo_warning
+
+REVISION = '1.0.15'
+DOWN_REVISION = '1.0.14'
+
+
+def notify_user(apps, schema_editor):
+    echo_warning("Invalidating all the hashes of all the nodes. Please " "run verdi rehash", bold=True)
```
Comment (sphuber) [inline aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py]: Obsolete double quotes (twice) in single string

File: aiida/backends/sqlalchemy/migrations/versions/5d4d844852b6_invalidating_node_hash.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+down_revision = '62fe0d36de90'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+
+    # Invalidate all the hashes & inform the user
+    echo_warning("Invalidating all the hashes of all the nodes. " "Please run verdi rehash", bold=True)
```
Comment (sphuber) [inline aiida/backends/sqlalchemy/migrations/versions/5d4d844852b6_invalidating_node_hash.py]: unnecessary double quotes

File: aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py (hunk 1 above)
Comment (szoupanos) [reply aiida/backends/djsite/db/migrations/0015_invalidating_node_hash.py]: That's a good point. Thanks

**Pull request merged**